### ISSUE TITLE
Disable automatic hyperlinking on a per-notebook and per-cell basis

### ISSUE THREAD

**Issue by muraiki** [jupyterlab/jupyterlab#10704]
### Problem

I would like it if there would be a way to disable automatic hyperlinking in cell output. In particular, when working with cybersecurity data, it can be dangerous to users if cell output containing data with malicious URLs is automatically hyperlinked.

### Proposed Solution

There should be a way to disable this for an entire notebook and for a specific cell. Perhaps this can be done using magics? If this is instead done using a user preference, that would still expose consumers of the notebook code to being able to accidentally click on malicious URLs.

### Additional context

In cybersecurity, sometimes URLs are formatted in a special way as a safeguard against this, such as replacing `.` with `,` or `[.]` (see [Defanging URLs](https://www.ibm.com/docs/en/rsoa-and-rp/32.0?topic=SSBRUQ_32.0.0/com.ibm.resilient.doc/install/resilient_install_defangURLs.htm) for examples), as automatic hyperlinking is common in programs. While one could do this as part of the data cleaning process, the URLs then can't be parsed with URL-parsing libraries without removing that formatting for every operation.

**Comment by MMesch** [comment]
I came across this because the triaging agent identified this as potentially resolved, and I think I agree:

- The requested *global* functionality has already been implemented in JupyterLab. Users can disable automatic URL hyperlinking globally through the HTML Sanitizer settings in the Settings Editor.
- Per-notebook and per-cell basis control is *not* implemented in core JupyterLab but
  - wouldn't it best be done as you said yourself, with a function that defangs individual URLs, or multiple URLs within a longer string if you need to capture stdout/stderr etc? I don't see why this would impede capturing the string with raw urls independently for further parsing. Using a function has the advantage of being independent of the particular Jupyterlab configuration in which a notebook is executed. It seems there is a defang library available on pip.
  - Otherwise, I think implementing the cell-wise / notebook-wise option as extension but not in JupyterLab core would be the right way forward.

A related discussion is on discourse:

https://discourse.jupyter.org/t/is-there-any-way-to-disable-auto-generated-hyperlinks-in-cell/13594/2

Other opinions? If there is agreement on the above viewpoint I suggest closing this as not planned.

**Comment by brichet** [comment]
@MMesch I agree that the main expectation of this issue is already fulfilled in core Jupyterlab, with the `HTML Sanitizer` settings.

The per-cell/per-notebook control should be implemented in an extension, and would probably require some metadata in the notebook to be a persistent across users/installations.

We can close this issue as not planned.

**Comment by krassowski** [comment]
Well, we already have trusted notebook status. Maybe there should be a setting "Do not auto-link in untrusted notebooks"? I think this would be a good enhancement without a need to store metadata.

**Comment by MMesch** [comment]
That sounds quite useful to me as well but we should rename the issue then or maybe open a new one.
